testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorSmokeTest.java

```diff
@@ -116,12 +124,18 @@ public void testAggregation()
     @Test
     public void testHaving()
     {
+        if (!hasBehavior(SUPPORTS_HAVING_CLAUSE)) {
+            return;
+        }
         assertQuery("SELECT regionkey, sum(nationkey) FROM nation GROUP BY regionkey HAVING sum(nationkey) = 58", "VALUES (4, 58)");
     }
 
     @Test
     public void testJoin()
     {
+        if (!hasBehavior(SUPPORTS_JOIN_PUSHDOWN)) {
+            return;
+        }
```

Comment:
Please revert, this test isn't for JOIN **_pushdown_**.

Reply:
Thanks for pointing out this doesn't fall under JOIN pushdown. Before I change this, this test of JOIN fails with Pinot queries. So would you suggest these queries be ignored or anything else?